src/Microsoft.Azure.Functions.Worker.Extensions.Mcp/Converters/TypeConverter.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Text.Json;
+using Microsoft.Azure.Functions.Worker.Converters;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.Mcp.Converters;
+
+internal class TypeConverter : IInputConverter
```

Comment:
Added to question from @mattchenderson , but also wanted to leave this one comment here: we should use a more descriptive type name here. `TypeConverter` is a bit generic and we should consider something that more specifically describes this converter performs argument binding/conversion.